src/Data/Barbie/Internal/Constraints.hs

```diff
@@ -49,46 +50,68 @@ import GHC.Generics
 -- data T f = A (f 'Int') (f 'String') | B (f 'Bool') (f 'Int')
 --
 -- instance 'ConstraintsB' T where
---   type 'ConstraintsOf' c f T
---     = (c (f 'Int'), c (f 'String'), c (f 'Bool'))
+--   type 'AllB' c T
+--     = (c 'Int', c 'String', c 'Bool')
+--
+--   type 'NotBare' T f
+--     = ()
 --
 --   adjProof t = case t of
 --     A x y -> A ('Pair' ('packDict' x) ('packDict' y))
 --     B z w -> B ('Pair' ('packDict' z) ('packDict' w))
 -- @
 --
--- There is a default implementation of 'ConstraintsOf' for
+-- There is a default implementation of 'AllB' for
 -- 'Generic' types, so in practice one will simply do:
 --
 -- @
 -- derive instance 'Generic' T
 -- instance 'ConstraintsB' T
 -- @
 class FunctorB b => ConstraintsB b where
-  -- | @'ConstraintsOf' c f b@ should contain a constraint @c (f x)@
-  --  for each @f x@ occurring in @b@. E.g.:
+  -- | @'AllB' c b@ should contain a constraint @c x@ for each
+  --   @f x@ occurring in @b@. E.g.:
```

Comment:
This should say "for each @x@", right?

Reply:
Not sure. I mean, let's consider this case:
```
data T f = T (f Int) Bool
```
With the code as it is we would then have
```
AllB c T = c Int
```
So in that sense `for each @f x@`  seems to be more correct; maybe this can be made more clear?

There is then question of whether the definition of `AllB` is ok as it is or it should actually be
```
AllB c T = (c Int, c Bool)
```
Not sure about this one. I think in practice nothing would break if we use the latter, but I can't think right away of anything we would be able to do with the additional constraint.